templates/Template.NEP5.CSharp/NEP5.Methods.cs

```
@@ -1,51 +1,50 @@
 using System;
+using System.ComponentModel;
 using System.Numerics;
 
 namespace Template.NEP5.CSharp
 {
     public partial class NEP5 : SmartContract
     {
-        public static BigInteger TotalSupply()
-        {
-            StorageMap contract = Storage.CurrentContext.CreateMap(StoragePrefixContract);
-            return contract.Get("totalSupply")?.ToBigInteger() ?? 0;
-        }
+        public static BigInteger TotalSupply() => TotalSupplyStorage.Get();
 
-        public static BigInteger BalanceOf(byte[] account)
-        {
-            if (!ValidateAddress(account)) throw new FormatException("The parameter 'account' SHOULD be 20-byte addresses.");
-
-            StorageMap balances = Storage.CurrentContext.CreateMap(StoragePrefixBalance);
-            return balances.Get(account)?.ToBigInteger() ?? 0;
-        }
+        public static BigInteger BalanceOf(byte[] account) => AssetStorage.Get(account);
 
         public static bool Transfer(byte[] from, byte[] to, BigInteger amount)
         {
-            if (!ValidateAddress(from)) throw new FormatException("The parameter 'from' SHOULD be 20-byte addresses.");
-            if (!ValidateAddress(to)) throw new FormatException("The parameters 'to' SHOULD be 20-byte addresses.");
-            if (!IsPayable(to)) return false;
-            if (amount <= 0) throw new InvalidOperationException("The parameter amount MUST be greater than 0.");
-            if (!Runtime.CheckWitness(from)) return false;
-
-            StorageMap balances = Storage.CurrentContext.CreateMap(StoragePrefixBalance);
-            BigInteger fromAmount = balances.Get(from).ToBigInteger();
-
-            if (fromAmount < amount) return false;
-            if (amount == 0 || from == to) return true;
-
-            if (fromAmount == amount)
+            if (!ValidateAddress(from) || !ValidateAddress(to))
             {
-                balances.Delete(from);
+                Error("The parameters from and to SHOULD be 20-byte addresses.");
```

No, should keep it, throw an exception. This is defined in nep5 standard.

_The parameters from and to SHOULD be 20-byte addresses. If not, this method SHOULD throw an exception._